firmware-binaries/clock-board/src/main.rs

```diff
@@ -0,0 +1,150 @@
+#![no_std]
+#![cfg_attr(not(test), no_main)]
+
+// SPDX-FileCopyrightText: 2025 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use core::panic::PanicInfo;
+use ufmt::derive::uDebug;
+use ufmt::uwriteln;
+
+use bittide_hal::manual_additions::si539x_spi::Config;
+use bittide_hal::manual_additions::timer::Duration;
+use bittide_hal::shared::devices::Timer;
+use bittide_hal::shared::devices::Uart;
+use bittide_hal::si539x_configuration::DeviceInstances;
+use bittide_hal::si539x_configuration::DomainDiffCounters;
+
+use bittide_macros::load_clock_config_csv;
+
+#[cfg(not(test))]
+use riscv_rt::entry;
+
+#[derive(uDebug, Clone, Copy)]
+enum FrequencyCheckError {
+    CounterInactive,
+    CounterInaccessible,
+    UnexpectedFrequency,
+}
+
+const INSTANCES: DeviceInstances = unsafe { DeviceInstances::new() };
+const CONFIG_100: Config<3, 590, 5> =
+    load_clock_config_csv!("../../bittide/data/clock_configs/Si5395J-100MHz-10ppb-and-out1.csv");
+const CONFIG_200: Config<3, 584, 5> =
+    load_clock_config_csv!("../../bittide/data/clock_configs/Si5395J-200MHz-10ppb-and-out1.csv");
```

Comment:
We discussed offline to not add new configurations.

It's theoretically possible to get a false positive on the write behavior of the core if the HITL test prior to this one wrote the same configuration.

To check if we can write registers using this core we're going to modify the design ID register and read back the changed values.